algebra2/gaussian_elimination/README.md

```diff
@@ -0,0 +1,30 @@
+# Project Description
+
+Our project allows end-users to perform Gaussian elimination on matrices of different sizes. It's based on three basic algorithms: swapping two rows, multiplying a row by a nonzero number, and adding a multiple of one row to another row.
+
+## How to Run the Code
+To compile the code on your computer, navigate to the build directory and use the following command:
+
+clang++-18 -stdlib=libc++ -std=c++23 ./main.cpp -o main
+
+**NOTE**: Ensure that the files `gaussian_elimination.h`, `matrix.hpp`, and `main.cpp` are in the same directory.
+
+## Functions
+
+- `fraction()` allows performing tasks on fractions, e.g., `1/8`.
+- `run()` utilizes the three operations mentioned in the Project Description.
+- `show_steps()` prints out the matrix between using two operations.
+- `determinant()` calculates the determinant of the matrix if the matrix is square.
+- `inverse()` calculates the inverse matrix of matrix `m` if the matrix is square and invertible.
+
+You can look at `gaussian_elimination_examples()` and `gaussian_elimination_examples_fractions()` to understand how to declare matrices and call functions depending on if you need to operate on fractions or floating point numbers
+
+## Questions
+If there's a part of the code not mentioned in this document that needs explanation, please refer to the source code or ask the authors.
+
+## Authors
```

Comment:
You can sign yourselves but this repo is public so you should be aware that your names will be published (maybe change your names to github nicks or sth like that? - it is your decision)